### PR TITLE
Abstraction layer

### PR DESCRIPTION
**Pull request by DrItanium**
Type sizes are now abstracted (hopefully) enough that porting or building for new targets isn't a total chore.

### CODE REVIEW

**Comment by DanielGibson** [comment]
MSVC 2010 and newer support stdint.h.
As the Serious Engine wants MSVC 2013 or 2015, it should be safe to use it, so you could get rid of some Windows special cases.

**Comment by icculus** [comment]
@DanielGibson This is cool to merge with the PR I just took in, right?

**Comment by DanielGibson** [comment]
Should be, yes

**Comment by icculus** [comment]
Done, thank you everyone!

File: Sources/Engine/Base/Base.h [hunk 1]
```diff
@@ -51,8 +53,29 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
-#endif  // defined _MSC_VER
+#elif (defined __linux__) 
+  #if (defined __ANDROID__) || (defined __android__) 
+    #error "Android current isn't supported"
+  #else
+    #define PLATFORM_LINUX
+  #endif
+#elif (defined __APPLE__)
+  #include "TargetConditionals.h"
+  #if TARGET_OS_MAC
```
Comment (DanielGibson) [inline Sources/Engine/Base/Base.h:65]: this doesn't work properly, as `Source/Engine/StdH.h` does `#undef TARGET_OS_MAC`

Comment (DrItanium) [inline Sources/Engine/Base/Base.h:65]: Okay, I'll fix this issue.

**Pull request merged**